packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/create-model-customization-job.ts

```diff
@@ -0,0 +1,486 @@
+          },
+        },
+      }),
+      new iam.PolicyStatement({
+        actions: ['ec2:CreateNetworkInterface'],
+        resources: [
+          ...vpcConfig.securityGroups.map(
+            (sg) => Stack.of(this).formatArn({
+              service: 'ec2',
+              resource: 'security-group',
+              resourceName: sg.securityGroupId,
+            })),
+          ...vpcConfig.subnets.map(
+            (subnet) => Stack.of(this).formatArn({
+              service: 'ec2',
+              resource: 'subnet',
+              resourceName: subnet.subnetId,
+            }),
+          ),
+        ],
+      }),
+      new iam.PolicyStatement({
+        actions: [
+          'ec2:CreateNetworkInterfacePermission',
+          'ec2:DeleteNetworkInterface',
+          'ec2:DeleteNetworkInterfacePermission',
+        ],
+        resources: ['*'],
+        conditions: {
+          ArnEquals: {
+            'ec2:Subnet': vpcConfig.subnets.map(
+              (subnet) => Stack.of(this).formatArn({
+                service: 'ec2',
+                resource: 'subnet',
+                resourceName: subnet.subnetId,
+              })),
+            'ec2:ResourceTag/BedrockModelCustomizationJobArn': [
+              Stack.of(this).formatArn({
+                service: 'bedrock',
+                resource: 'model-customization-job',
+                resourceName: '*',
+              }),
+            ],
+          },
+          StringEquals: {
+            'ec2:ResourceTag/BedrockManaged': 'true',
+          },
+        },
+      }),
+      new iam.PolicyStatement({
+        actions: ['ec2:CreateTags'],
+        resources: [
+          Stack.of(this).formatArn({
+            service: 'ec2',
+            resource: 'network-interface',
+            resourceName: '*',
+          }),
+        ],
+        conditions: {
+          'StringEquals': {
+            'ec2:CreateAction': ['CreateNetworkInterface'],
+          },
+          'ForAllValues:StringEquals': {
+            'aws:TagKeys': [
+              'BedrockManaged',
+              'BedrockModelCustomizationJobArn',
+            ],
+          },
+        },
+      }),
+    ];
+  }
+
+  private renderPolicyStatements(): iam.PolicyStatement[] {
+    const policyStatements = [
+      new iam.PolicyStatement({
+        actions: [
+          'bedrock:CreateModelCustomizationJob',
+          'bedrock:TagResource',
+        ],
+        resources: [
+          this.props.baseModel.modelArn,
+          Stack.of(this).formatArn({
+            service: 'bedrock',
+            resource: 'custom-model',
+            resourceName: '*',
+          }),
+          Stack.of(this).formatArn({
+            service: 'bedrock',
+            resource: 'model-customization-job',
+            resourceName: '*',
+          }),
+        ],
+      }),
+      new iam.PolicyStatement({
+        actions: ['iam:PassRole'],
+        resources: [this._role.roleArn],
+      }),
+      ...(this.props.customModelKmsKey
```

Comment:
Following point 2, this should be attached to the `customModelKmsKey` key policy.

Reply:
You are right. I added resource based policy to `customModelKmsKey`

```ts
if (this.props.customModelKmsKey) {
      this.props.customModelKmsKey.addToResourcePolicy(new iam.PolicyStatement({
        actions: ['kms:Decrypt', 'kms:GenerateDataKey', 'kms:DescribeKey', 'kms:CreateGrant'],
        resources: ['*'],
        principals: [new iam.ArnPrincipal(this._role.roleArn)],
      }));
    }
```